### DISCUSSION TITLE
Problema VScode su HA core su docker

### DISCUSSION THREAD

**Discussion by GrupponeMissionario** [indomus/forum#927] [GENERALE - Discussioni generiche]
Ciao a tutti,
cercavo di modificare i file .yaml da altro pc con windows usando VSCode.
Ho seguito la guida: 
[https://indomus.it/progetti/modificare-i-file-di-configurazione-home-assistant-core-su-docker-su-raspberry-con-visual-studio-code-vscode/](https://indomus.it/progetti/modificare-i-file-di-configurazione-home-assistant-core-su-docker-su-raspberry-con-visual-studio-code-vscode/)
Che sembra perfetta per la mia situazione (Raspberry Pi4 con Raspbian OS, Home Assistant Core su docker).
Riesco ad aprire i file, modificarli, ma non li salva. Da un errore di permessi.
Rispetto alla guida i miei file non si trovano sotto /home/pi/homeassistant/ ma si trovano su /home/homeassistant/.
Per l'installazione avevo seguito una guida sempre su inDomus.
Grazie.

**Comment by snakuzzo** [comment]
Adegua i permessi in modo tale da poterli modificare con l'utente con cui ti colleghi da remoto e/o aggiungi l'utente con cui ti colleghi da remoto al gruppo dell'utente proprietario dei file di home assistant

**Reply by GrupponeMissionario** [reply]
da remoto faccio il login con utente pi... che è lo stesso che uso da terminale in locale... quindi non capisco come mai... come verifico i permessi dell'utente?

**Reply by snakuzzo** [reply]
però perdonami, queste cose sono alla base di linux...non posso dirti o insegnarti io come si verificano i permessi dell'utente su un sistema operativo.
Se cerchi online trovi tutto

**Reply by GrupponeMissionario** [reply]
Mi sono rivolto a voi perchè ho installato HA seguendo una vostra guida passo passo.
Ho provato a configurare VSCode seguendo una vostra guida passo passo.
Allora chiedo: come mai la guida seguita da me prevedeva che l'utente homeassistant fosse separato dall'utente pi, invece ora la guida prevede che homeassistant venga installato sotto pi? Cosa cambia? la configurazione precedente creava dei problemi? come si fa eventualmente a spostare homeassistant?

**Reply by snakuzzo** [reply]
Tu puoi installare home assistant dove ti pare. Il percorso può essere pure `/pippo/pluto/paperino/`. Le guide vanno anche capite e non seguite solo passo passo come fosse un manuale. Io non posso darti una mano andando ad occhio dando per scontato la tua situazione sul tuo server. Se magari posti qualche dettaglio  in più, se ne può parlare

**Reply by GrupponeMissionario** [reply]
Quali dettagli ti potrebbero servire? non mi pareva giusto scrivere cose a caso... se sai che informazioni ti potrebbero servire per aiutarmi secondo me si fa prima.
Grazie.
EDIT: e comunque non mi torna che da ssh riesco a modificare i file con utente pi, e anche da app RaspController riesco con utente pi... e anche dal raspberry stesso da terminale uso l'utente pi per modificarli.... Infatti da VSCode se modifico i file usando il terminale funziona, solo con "open file" non funziona...
Appare questo:
`Non è stato possibile salvare 'automations.yaml': Non è possibile scrivere il file 'vscode-remote://ssh-remote+192.168.1.200/home/homeassistant/.homeassistant/automations.yaml' (NoPermissions (FileSystemError): Error: EACCES: permission denied, open '/home/homeassistant/.homeassistant/automations.yaml')`

**Reply by snakuzzo** [reply]
Ma non avevi detto che avevi installato Home Assistant sotto la home dell'utente `pi` ?
Io dall'ultimo messaggio vedo che provi a modificare i file sotto la home dell'utente `homeassistant` e credo sia normale che ti dia errore.
Ad ogni modo vanno sempre visti i permessi che hai impostato:

`ls -lart /home/homeassistant/.homeassistant`

**Reply by GrupponeMissionario** [reply]
[ls -lart.txt](https://github.com/indomus/forum/files/7398202/ls.-lart.txt)

Ecco cosa esce.
come scritto nel primo messaggio home assistant è installato su /home/homeassistant/.homeassistant
come dicevo nell'ultimo, da terminale con user pi riesco a modificare i file.

**Reply by snakuzzo** [reply]
Ho visto, ma secondo te come potrebbe mai funzionare la modifica dei file con altro utente se quei file sono tutti di `root` e può modificarli solo `root` ?
Non puoi modificare i file con l'utente `pi`...o meglio...puoi farlo solo usando `sudo`, che si tradurrebbe nel modificarli comunque come utente `root`.
Queste però sono le basi di Linux, se non sai nemmeno queste cose è difficile andare avanti e camminare con le proprie gambe. Inoltre la guida è una guida e non un manuale di istruzioni. Noi cerchiamo di essere il più precisi possibile, ma l'utente deve comunque avere una certa padronanza del proprio sistema...anche perché non possiamo sapere le cose che fa un utente sul proprio server.

Detto ciò...se non usi più l'utente homeassistant, ti basta cambiare l'ownership di quei file...

`sudo chown -R pi:pi /home/homeassistant/.homeassistant`

Fatto questo vedrai che con l'utente `pi` potrai modificarli.

Ad ogni modo, piccola aggiunta, non è corretto avere "roba" di `pi` sotto la home di un altro utente.
Ti consiglierei quindi, dopo aver cambiato l'ownership, di:

- Fermare il container docker di Home Assistant
- Spostare la dir `.homeassistant` sotto `/home/pi` (magari cambiando il nome togliendo quel "punto" iniziale, usando `cp` e non `mv` così nel caso lasci un backup dove è attualmente presente la directory):
Quindi...

`cp -R /home/homeassistant/.homeassistant /home/pi/homeassistant`

- Ricreare il container docker mappando la directory con `/config`

`-v /home/pi/homeassistant:/config`

**Reply by GrupponeMissionario** [reply]
Quello che non mi torna è che era la guida su indomus che faceva creare l'utente homeassistant, e faceva installare HA core su un percorso dedicato, lasciando stare l'utente pi.
Adesso quello che mi dici sembra logico e darebbe una bella sistemata togliendo passaggi inutili, ma come mai veniva consigliato di fare un utente dedicato?
Grazie

**Reply by snakuzzo** [reply]
Quella era una guida fatta secoli fa...solitamente è consigliabile avere un utente applicativo dedicato, ma in questo caso di fatto non è necessario
